### PR TITLE
[Translation] Fix handling of null messages in `ArrayLoader`

### LINKED ISSUE #49894: [Translation] `MessageCatalogue::add()` accepts non-string messages
### Symfony version(s) affected

5.4.22

### Description

`MessageCatalogue::set()` expects message to be string, but `MessageCatalogue::add()` (and the same applies for constructor) does not have any validation for messages array and in practice message could be anything, for example `null`. This creates inconsistencies, because `MessageCatalogue::all()` may return these keys with `null` as value even if `MessageCatalogue::has()` will return `false` for them.

In my case loading https://github.com/rob006-software/flarum-ext-last-post-avatar/blob/1ab3d5a5a7b8959f77481e389a1aba08bf97d37f/locale/en.yml resulted `rob006-last-post-avatar.forum` key with `null` as value, which resulted deprecation warning in Flarum, because they do not expected that `MessageCatalogue::all()` may return non-string messages (see https://github.com/rob006-software/flarum-ext-last-post-avatar/pull/2).

### How to reproduce

```php
$catalogue = new MessageCatalogue('en');
$catalogue->add(['test' => null]);
assert($catalogue->all() === [] || $catalogue->all(['test' => ''])); 
```

### Possible Solution

I think that all messages should be cast to string before adding to `MessageCatalogue::$messages`. I case of loading YAML like https://github.com/rob006-software/flarum-ext-last-post-avatar/blob/1ab3d5a5a7b8959f77481e389a1aba08bf97d37f/locale/en.yml I would expect that these null-keys will be completely skipped, but I'm not sure if skipping them in `MessageCatalogue::add()` would be a good place to do this.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by rob006**
| Q             | A
| ------------- | ---
| Branch?       |  5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #49894
| License       | MIT
| Doc PR        | -

Replaces https://github.com/symfony/symfony/pull/50252

### CODE REVIEW

File: src/Symfony/Component/Translation/Loader/ArrayLoader.php [hunk 1]
```diff
@@ -46,10 +46,12 @@ private function flatten(array $messages): array
...
                 foreach ($this->flatten($value) as $k => $v) {
-                    $result[$key.'.'.$k] = $v;
+                    if (null !== $v) {
+                        $result[$key.'.'.$k] = (string) $v;
+                    }
                 }
-            } else {
-                $result[$key] = $value;
+            } elseif (null !== $value) {
+                $result[$key] = (string) $value;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Translation/Loader/ArrayLoader.php]: I think we should remove the cast to string, so that stringable objects aren't resolved eagerly, if that makes sense

Comment (rob006) [reply src/Symfony/Component/Translation/Loader/ArrayLoader.php]: I don't think that `MessageCatalogue` expects non-string input - in 6.x line there are already types that would force cast to string in `get()` and `set()`. If performance is concern, then cache should be used, and stringable objects here would only be a problem in that case.

Comment (nicolas-grekas) [reply src/Symfony/Component/Translation/Loader/ArrayLoader.php]: Still: your PR is motivated by nulls, not by non-strings. Since this is 5.4, we'd better not change the behavior unless we have a compelling reason to.

**Comment by nicolas-grekas** [comment]
Thank you @rob006.
